src/main/java/fi/aalto/cs/apluscourses/presentation/ideactivities/TutorialViewModel.java

```diff
@@ -52,6 +58,7 @@ public TutorialViewModel(@NotNull TutorialExercise tutorialExercise,
     currentTaskIndex = 1;
     unlockedIndex = 1;
     tasksAmount = tasks.size();
+    tutorialWindowFrame = JOptionPane.getRootFrame();
```

Comment:
This gets the focused window frame at the moment, right? I don't think it's safe to assume that at the time this object is created, the related window is necessarily focused.

Reply:
Agreed, this will be removed anyway.